src/extras/primitives/registerPrimitive.js

```diff
@@ -72,16 +72,19 @@ module.exports = function registerPrimitive (name, definition) {
         value: function () {
           var self = this;
           var defaultData = this.defaultComponentsFromPrimitive;
+
           // Apply default components.
           Object.keys(defaultData).forEach(function applyDefault (componentName) {
             var componentData = defaultData[componentName];
+
             // Set component properties individually to not overwrite user-defined components.
-            if (componentData instanceof Object && Object.keys(componentData).length) {
+            if (componentData instanceof Object) {
```

Comment:
It shouldn't have been there in the first place. It was there thinking that default components would only overwrite values if they had properties attached, but they were overwriting values regardless.